metrics/error.global/error.go

```diff
@@ -0,0 +1,305 @@
+				},
+			},
+		},
+		Groups: []blip.CollectorKeyValue{
+			{Key: GRP_ERR_NUMBER, Value: "the error number, or an empty string for a total"},
+			{Key: GRP_ERR_NAME, Value: "the error name, or an empty string for a total"},
+		},
+		Metrics: []blip.CollectorMetric{
+			{
+				Name: "raised",
+				Type: blip.DELTA_COUNTER,
+				Desc: "Raised errors",
+			},
+		},
+		Errors: map[string]blip.CollectorHelpError{
+			ERR_TRUNCATE_FAILED: {
+				Name:    ERR_TRUNCATE_FAILED,
+				Handles: "Truncation failures on error summary tables",
+				Default: errors.NewPolicy("").String(),
+			},
+		},
+	}
+}
+
+// Prepare prepares the collector for the given plan.
+func (t *ErrorGlobal) Prepare(ctx context.Context, plan blip.Plan) (func(), error) {
+LEVEL:
+	for _, level := range plan.Levels {
+		dom, ok := level.Collect[DOMAIN]
+		if !ok {
+			continue LEVEL // not collected in this level
+		}
+		if dom.Options == nil {
+			dom.Options = make(map[string]string)
+		}
+
+		if _, ok := dom.Options[OPT_ALL]; !ok {
+			dom.Options[OPT_ALL] = "yes"
+		}
+
+		if _, ok := dom.Options[OPT_TOTAL]; !ok {
+			dom.Options[OPT_TOTAL] = "yes"
+		}
+
+		errOpts, err := ErrorsQuery(dom)
+		if err != nil {
+			return nil, err
+		}
+
+		if total := dom.Options[OPT_TOTAL]; total == "only" || total == "no" {
+			errOpts.emitTotal = false
+		} else {
+			errOpts.emitTotal = true
+		}
+
+		if truncate, ok := dom.Options[OPT_TRUNCATE_TABLE]; ok && truncate == "yes" {
+			errOpts.truncate = true
+			errOpts.metricType = blip.DELTA_COUNTER
+
+		} else {
+			errOpts.truncate = false // default
+			errOpts.metricType = blip.CUMULATIVE_COUNTER
+		}
+
+		if truncateTimeout, ok := dom.Options[OPT_TRUNCATE_TIMEOUT]; ok && errOpts.truncate {
+			if duration, err := time.ParseDuration(truncateTimeout); err != nil {
+				return nil, fmt.Errorf("invalid truncate duration: %v", err)
+			} else {
+				errOpts.truncateTimeout = duration
+			}
+		} else {
+			errOpts.truncateTimeout = 250 * time.Millisecond // default
+		}
+
+		truncateOnStartup := true
+		if truncate := dom.Options[OPT_TRUNCATE_ON_STARTUP]; truncate == "no" {
+			truncateOnStartup = false
+		}
+
+		if errOpts.truncate || truncateOnStartup {
+			// Setup our lock wait timeout. It needs to be at least as long
+			// as our truncate timeout, but the granularity of the lock wait
+			// timeout is seconds, so we round up to the nearest second that is
+			// greater than our truncate timeout.
+			lockWaitTimeout := math.Ceil(errOpts.truncateTimeout.Seconds())
+			if lockWaitTimeout < 1.0 {
+				lockWaitTimeout = 1
+			}
+
+			errOpts.lockWaitQuery = fmt.Sprintf(LOCKWAIT_QUERY, int64(lockWaitTimeout))
+			errOpts.truncateErrPolicy = errors.NewTruncateErrorPolicy(dom.Errors[ERR_TRUNCATE_FAILED])
+			blip.Debug("error policy: %s=%s", ERR_TRUNCATE_FAILED, errOpts.truncateErrPolicy.Policy)
+		}
+
+		t.options[level.Name] = errOpts
+
+		// Run an initial truncate to clear out any old data
```

Comment:
Not sure `Prepare` is right place for this... hard to say. Is truncate on startup a setup (prepare) thing? Or part of collection (i.e. a pre-step and requirement to `Collect`)? Hm...